src/libraries/Microsoft.Extensions.Hosting/src/HostOptions.cs

```diff
@@ -17,6 +17,15 @@ public class HostOptions
         /// </summary>
         public TimeSpan ShutdownTimeout { get; set; } = TimeSpan.FromSeconds(30);
 
+        /// <summary>
+        /// The behavior the <see cref="IHost"/> will follow when stopping registered instances of <see cref="IHostedService"/>
+        /// </summary>
+        /// <remarks>
+        /// Defaults to <see cref="BackgroundServiceStopBehavior.Asynchronous"/>
+        /// </remarks>
+        public BackgroundServiceStopBehavior BackgroundServiceStopBehavior { get; set; } =
```

Comment:
Public API changes need to go through https://aka.ms/api-review process.

Reply:
> We will need to add some tests for the new and old behavior.

I agree, but I just wanted to see if this solution is the route that would be preferred prior to spending time on tests.

Reply:
> Public API changes need to go through https://aka.ms/api-review process.

It looks like the issue still needs a sponsor.

Reply:
Can you update the issue to follow the [template](https://github.com/dotnet/runtime/issues/new?assignees=&labels=api-suggestion&template=02_api_proposal.yml&title=%5BAPI+Proposal%5D%3A+)?

Reply:
Done